cmd/jujud/agent/agenttest/engine.go

```diff
@@ -39,6 +39,14 @@ type WorkerMatcher struct {
 	id        string
 	expect    set.Strings
 	matchTime time.Time
+	optional  set.Strings
+}
+
+// AddOptionalWorkers adds workers which may or maynot start.
+// Use sparingly. Intended when these unit tests have different
+// results based on hardware.
+func (m *WorkerMatcher) AddOptionalWorkers(optional []string) {
+	m.optional = set.NewStrings(optional...)
```

Comment:
The keyword is here is "Add" where the operation is more of a set i.e wipe what is already there. Do we want to change the name or functionality?

Reply:
My thought was you are adding data to the matcher which cannot be set in the constructor. I didn't want to change the constructer in many places in test.